### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by malnick**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+What is the change? 
+Why is the change needed?
+How does this change affect the user experience (if at all)?
+
+# Design of Change
+How was this change implemented?
+
+# Related Issues/Pull Requests
+[ ] [Issue #1234](https://github.com/hashicorp/vault/issues/1234)
+[ ] [PR #1234](https://github.com/hashicorp/vault/pr/1234)
```
Comment (pcman312) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm not sure why, but this is showing up on the same line as the previous checkbox in the rich view

Comment (pcman312) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think this needs to be in an unordered list:
```
- [ ] ...
```

Comment (kalafut) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think it's just the preview that's off

Comment (pcman312) [reply .github/PULL_REQUEST_TEMPLATE.md]: It looks off when I view the file directly: https://github.com/hashicorp/vault-plugin-auth-alicloud/blob/ccfabfd502c5d7edb663511e90f176ebcb677675/.github/PULL_REQUEST_TEMPLATE.md


Screen Shot 2020-03-30 at 4 41 21 PM

Comment (kalafut) [reply .github/PULL_REQUEST_TEMPLATE.md]: But it does look OK when filling out the PR AFAICT.

**Pull request merged**